vtex-sites/gatsby.store
# Chore: storybook improvements

Ported from https://github.com/vtex-sites/nextjs.store/pull/129

## What's the purpose of this pull request?
General improvements on Storybook.

## How does it work?

- Displays the `Sandbox` tab: so we can test viewports and use other addons (Accessibility)
- Adds project's viewports to help with components' resizes tests
![preview-tab](http://g.recordit.co/aRRoURLqx6.gif)
- Fixes some console errors:
      -   [Importing from '@storybook/addon-docs/blocks' is deprecated. ](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md#deprecated-scoped-blocks-imports)
      - Adds `OnChange` handler to components examples (`Checkbox`, `Radio`, `Toggle`)
![image](https://user-images.githubusercontent.com/3356699/174214466-efc88c3b-65b1-4c46-aacc-314b3f0efd52.png)

## How to test it?

- Check on the [preview](https://gatsby-store-storybook-git-chore-storybook-imp-b33798-faststore.vercel.app/?path=/docs/getting-started--page) if `Sandbox` and `Docs` tabs are available (except `Getting Started`, `Theming`, `Fundamentals` pages and `Changelog`)
- When switching tabs, `Sandbox` might not be working properly until you reload the page. 
 There is a note on the [Getting Started's](https://gatsby-store-storybook-git-chore-storybook-imp-b33798-faststore.vercel.app/?path=/docs/getting-started--page) page regarding this issue
![image](https://user-images.githubusercontent.com/3356699/174346416-cda2a289-ca6e-4846-a52d-b26797cf76cf.png)


## References

- [Canvas tab not working properly until after reload when using --docs flag. ](https://github.com/storybookjs/storybook/issues/17625)
- [Viewport Addon](https://storybook.js.org/docs/react/essentials/viewport)
- https://github.com/storybookjs/storybook/pull/9095

## Checklist

<em>You may erase this after checking them all ;)</em>

**Changelog**

**PR Description**

**Documentation**
